### PR TITLE
feat: upgrade "constructs" to 3.2.0

### PR DESCRIPTION
**Pull request by eladb**
This new version includes the `node.addr` property which deprecates `uniqueId`.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by rix0rrr — APPROVED** [review]

File: packages/monocdk/package.json [hunk 1]
```diff
@@ -256,7 +256,7 @@
     "@types/fs-extra": "^8.1.1",
     "@types/node": "^10.17.42",
     "cdk-build-tools": "0.0.0",
-    "constructs": "^3.0.4",
+    "constructs": "^3.2.0",
```
Comment (NetaNir) [inline packages/monocdk/package.json:259]: This is a braking change for monocdk.

Comment (NetaNir) [reply packages/monocdk/package.json:259]: added do-not-merge

Comment (NetaNir) [reply packages/monocdk/package.json:259]: I  was wrong, this is not a breaking change as the version bump is for the `devDependencies` and not for `peerDependencies`. But we should still keep the version in `devDependencies` to be the minimal version we support

Comment (eladb) [reply packages/monocdk/package.json:259]: Yes, this was an oversight that I only bumped the `devDependency` and not the peer. We need to discuss what's the path to allow ourselves to upgrade the minimal peer requirement. Otherwise we are stuck forever in 3.0.4 which does not make sense.

Comment (NetaNir) [reply packages/monocdk/package.json:259]: To add some background. Peer dependencies are a part of the public API. A consumer `package.json` could include:

```js
dependencies: {
    "monocdk": "^1.69.0",
     "constructs": "3.0.4"  
}
```

And it is a valid configuration. 

If `monocdk`  version `1.70.0` bumps the minimal supported version of `constructs` to `3.2.0` , when trying to upgrade to version `1.70.0` the consumer will break as it must bump the `constructs` version as well. There could be many reason for a consumer to declare a fixed dependency on `constructs. Could be because they are using an experimental API of constructs that was removed from newer versions, or they use a third party dependency that declares a direct dependency on the older version of `constructs`.

Comment (skinny85) [reply packages/monocdk/package.json:259]: 2 points:

1. Does `constructs` contain APIs that can break backwards compatibility in minor version upgrades? Because I'm pretty sure it can't, for the same reason [mono-CDK can't](https://github.com/aws/aws-cdk-rfcs/blob/master/text/0249-v2-experiments.md#appendix-a---why-cant-we-break-backwards-compatibility-in-the-code-of-mono-cdk-main-modules).

2. I don't think third-party packages should have a direct dependency on `constructs`, they should use peer dependencies as well.

In conclusion: I think it's OK to bump the minimum supported minor version of the `constructs` peer dependency. Not being able to do that would be extremely limiting IMO.

Comment (NetaNir) [reply packages/monocdk/package.json:259]: It's not about API breaking. It's about not being able to upgrade to a newer version of `monocdk` without upgrading `constructs`.

Comment (skinny85) [reply packages/monocdk/package.json:259]: 1. That's fine. If there are no backwards-incompatible changes possible in the minor version bump of `constructs`, that means they can always safely update the version of `constructs` that they use.
2. Sure. A third-party library can also depend on `constructs` version `3.0.3`, in which case we have this problem already. For that reason, I don't think it's worth it to spend a lot of time worrying about this particular problem.

**Review by rix0rrr — COMMENTED** [review]
The question at hand seems to be:

*Is forcing a user to upgrade a version of a related dependency when they upgrade yours a breaking change?*

**Morally:** it's a tough one, but I would think I'm erring on the side of "no". They're already in the process of upgrading: as long as they're using a lockfile (and they should be) it's a manual action to do that. As long as they don't have to change any of their code I think it's defensible to say that's just what it is.

**Practically:** is a commonly used automated system like Dependabot able to handle coordinated upgrades like these? If not it might be a bad idea to do them...

But in any case, I feel we kinda have to, otherwise we'll never be able to get anything done.

**Comment by eladb** [comment]
I agree we simply can't avoid minor-upgrading our peer requirement on `constructs` after we release v2.0. Based on what we recommend to customers, I think this is a reasonable "breakage".

For applications, we recommend that users declare a caret (`^`) dependency on both modules: `{ "constructs": "^3.0.4", "aws-cdk-lib": "^2.1.0" }`. In this case, if their package manager picks up a newer version of `aws-cdk-lib` which has a higher requirement for `constructs`, the package manager will bring in the newer version. So this is basically seamless and does not incur a break.

For libraries, we recommend that the library project declares dependencies like so:

```json
{
  "peerDependencies": {
    "constructs": "^3.0.4",
    "aws-cdk-lib": "^2.1.0"
  },
  "devDependencies": {
    "constructs": "3.0.4",
    "aws-cdk-lib": "2.1.0"
  }
}
```

The reason we recommend adding _pinned_ `devDependencies` is to ensure that the library is tested against the minimum required version of the peers. 

In the library case, bumping `aws-cdk-lib`'s requirement on `constructs` to `3.2.0` means that when the library author upgrades `aws-cdk-lib` they will have to always upgrade their devDependency (otherwise, there will be a mismatch) and consequently, their peer requirement.

As @rix0rrr wrote earlier, I think this is a reasonable experience. Given the library author explicitly decided to bump their CDK version (since they use a pinned version in the devDependency this will not happen in an unsolicited manner), it's fair to require that they now bump their `constructs` requirement as well in the same process.

File: packages/monocdk/package.json (hunk 1 above)
Comment (eladb) [reply packages/monocdk/package.json:259]: See https://github.com/aws/aws-cdk/pull/11145#issuecomment-717786931
